// react-social-bar/lib/SocialBox/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = SocialBox;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

require('./SocialBox.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propTypes = {
  children: _propTypes2.default.element,
  handleClick: _propTypes2.default.func,
  name: _propTypes2.default.string.isRequired
};

var defaultProps = {
  children: _react2.default.createElement('span', null)
};

function SocialBox(_ref) {
  var children = _ref.children,
      handleClick = _ref.handleClick,
      name = _ref.name;


  var getClass = function getClass() {
    switch (name) {
      case 'linkedin':
        return 'linkedin';
      case 'facebook':
        return 'facebook';
      case 'pinterest':
        return 'pinterest';
      case 'twitter':
        return 'twitter';
      default:
        return '';
    }
  };

  return _react2.default.createElement(
    'div',
    {
      className: 'SocialBox SocialBox-' + getClass(),
      onClick: handleClick
    },
    children
  );
}

SocialBox.propTypes = propTypes;
SocialBox.defaultProps = defaultProps;
