// react-social-bar/lib/SocialBar/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _SocialBox = require('../SocialBox');

var _SocialBox2 = _interopRequireDefault(_SocialBox);

var _Facebook = require('../Facebook');

var _Facebook2 = _interopRequireDefault(_Facebook);

var _LinkedIn = require('../LinkedIn');

var _LinkedIn2 = _interopRequireDefault(_LinkedIn);

var _Pinterest = require('../Pinterest');

var _Pinterest2 = _interopRequireDefault(_Pinterest);

var _Twitter = require('../Twitter');

var _Twitter2 = _interopRequireDefault(_Twitter);

require('./SocialBar.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var SocialBar = function (_Component) {
  _inherits(SocialBar, _Component);

  function SocialBar() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, SocialBar);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = SocialBar.__proto__ || Object.getPrototypeOf(SocialBar)).call.apply(_ref, [this].concat(args))), _this), _this._handleLinkedIn = function () {
      var link = 'https://www.linkedin.com/shareArticle?mini=true';
      var url = '&url=' + _this.props.domain;
      var title = '&title=' + _this.props.productName;
      var summary = '&summary=' + _this.props.description;
      var source = '&source=' + _this.props.domain;
      link = encodeURI(link + url + title + summary + source);

      _this.props.handleClick('linkedin');
      window.open(link, 'linkedin window', "height=450,width=650");
    }, _this._handleTwitter = function () {
      var link = 'https://twitter.com/intent/tweet?&via=' + _this.props.twitterHandle;
      var url = '&url=' + _this.props.domain;
      var message = '&text=' + _this.props.description;
      link = encodeURI(link + message + url);

      _this.props.handleClick('twitter');
      console.log('twitter link: ', link);
      window.open(link, 'twitter window', "height=420,width=650");
    }, _this._handlePinterest = function () {
      var url = 'http://pinterest.com/pin/create/button/?url=' + _this.props.domain;
      var description = '&description=' + _this.props.description;
      var media = '&media=' + _this.props.mediaUrl;
      url = encodeURI(url + description + media);

      _this.props.handleClick('pinterest');
      window.open(url, 'pinterest window', "height=625,width=770");
    }, _this._handleFacebook = function () {
      var url = 'https://www.facebook.com/sharer/sharer.php?u=' + _this.props.domain;

      _this.props.handleClick('facebook');
      window.open(url, 'facebook window', "height=420,width=650");
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  // PRIVATE

  _createClass(SocialBar, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        { className: 'SocialBar' },
        _react2.default.createElement(
          _SocialBox2.default,
          {
            name: 'linkedin',
            handleClick: this._handleLinkedIn
          },
          _react2.default.createElement(_LinkedIn2.default, null)
        ),
        _react2.default.createElement(
          _SocialBox2.default,
          {
            name: 'pinterest',
            handleClick: this._handlePinterest
          },
          _react2.default.createElement(_Pinterest2.default, null)
        ),
        _react2.default.createElement(
          _SocialBox2.default,
          {
            name: 'facebook',
            handleClick: this._handleFacebook
          },
          _react2.default.createElement(_Facebook2.default, null)
        ),
        _react2.default.createElement(
          _SocialBox2.default,
          {
            name: 'twitter',
            handleClick: this._handleTwitter
          },
          _react2.default.createElement(_Twitter2.default, null)
        )
      );
    }
  }]);

  return SocialBar;
}(_react.Component);

SocialBar.propTypes = {
  description: _propTypes2.default.string.isRequired,
  domain: _propTypes2.default.string.isRequired,
  handleClick: _propTypes2.default.func.isRequired,
  mediaUrl: _propTypes2.default.string,
  productName: _propTypes2.default.string.isRequired,
  twitterHandle: _propTypes2.default.string
};
exports.default = SocialBar;
